bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/handler/AbstractOmnilinkHandler.java

```diff
@@ -0,0 +1,91 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.omnilink.internal.handler;
+
+import static org.openhab.binding.omnilink.internal.OmnilinkBindingConstants.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.digitaldan.jomnilinkII.OmniInvalidResponseException;
+import com.digitaldan.jomnilinkII.OmniUnknownMessageTypeException;
+
+/**
+ * The {@link AbstractOmnilinkHandler} defines some methods that can be used across
+ * the many different things exposed by the OmniLink protocol
+ *
+ * @author Brian O'Connell - Initial contribution
+ */
+@NonNullByDefault
+public abstract class AbstractOmnilinkHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(AbstractOmnilinkHandler.class);
+
+    public AbstractOmnilinkHandler(Thing thing) {
+        super(thing);
+    }
+
+    public @Nullable OmnilinkBridgeHandler getOmnilinkBridgeHandler() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            return (OmnilinkBridgeHandler) bridge.getHandler();
+        } else {
+            return null;
+        }
+    }
+
+    protected void sendOmnilinkCommand(int message, int param1, int param2) {
+        try {
+            final OmnilinkBridgeHandler bridge = getOmnilinkBridgeHandler();
+            if (bridge != null) {
+                bridge.sendOmnilinkCommand(message, param1, param2);
+            } else {
+                logger.debug("Received null bridge while sending OmniLink command!");
+            }
+        } catch (OmniInvalidResponseException | OmniUnknownMessageTypeException | BridgeOfflineException e) {
+            logger.error("Could not send command to OmniLink Controller!", e);
+        }
+    }
+
+    /**
+     * Gets the configured number for a thing.
+     *
+     * @return Configured number for a thing.
+     */
+    protected int getThingNumber() {
+        return ((Number) getThing().getConfiguration().get(THING_PROPERTIES_NUMBER)).intValue();
+    }
+
+    /**
+     * Gets the configured number for a thing.
+     *
+     * @return Configured number for a thing.
+     */
+    protected int getAreaNumber() {
+        return ((Number) getThing().getConfiguration().get(THING_PROPERTIES_AREA)).intValue();
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        super.bridgeStatusChanged(bridgeStatusInfo);
+        if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+    }
```

Comment:
Actually the framework takes care of the Thing status when the bridge is offline. Is this method really necessary?